src/main.zig

```diff
@@ -1711,47 +1712,55 @@ fn buildOutputType(
             if (runtime_args_start) |i| {
                 try argv.appendSlice(all_args[i..]);
             }
-            // TODO On operating systems that support it, do an execve here rather than child process,
-            // when watch=false and arg_mode == .run
-            const child = try std.ChildProcess.init(argv.items, gpa);
-            defer child.deinit();
+            if (std.builtin.os.tag == .linux and arg_mode == .run and watch == false) { // TODO does other posix os support execve in zig stdlib?
```

Comment:
This can be simplified with:
```zig
if (std.builtin.os.tag != .windows and arg_mode == .run and !watch)
```

The if doesn't really need an `else` as you either replace the running process or exit with `process.exit`. This saves you from indenting the body below and simplifies a bit the code.